TheFoodieNetwork/app/src/main/java/com/fbu/thefoodienetwork/fragments/ComposeFragment.java

```diff
@@ -0,0 +1,198 @@
+package com.fbu.thefoodienetwork.fragments;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.RatingBar;
+import android.widget.Spinner;
+import android.widget.Toast;
+
+import androidx.fragment.app.Fragment;
+
+import com.fbu.thefoodienetwork.R;
+import com.fbu.thefoodienetwork.databinding.FragmentComposeBinding;
+import com.fbu.thefoodienetwork.models.ParseRestaurant;
+import com.fbu.thefoodienetwork.models.ParseReview;
+import com.fbu.thefoodienetwork.models.Restaurant;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+import org.parceler.Parcels;
+
+import java.util.List;
+
+public class ComposeFragment extends Fragment implements AdapterView.OnItemSelectedListener {
+    private static final String TAG = "ComposeFragment";
+    private static final String ARG_RESTAURANT = "selectedRestaurant";
+    private FragmentComposeBinding binding;
+    private Restaurant mRestaurant;
+    private Spinner spinner;
+    private static final int EVERYONE = 0;
+    private static final int FRIENDS = 1;
+    private boolean shareWithEveryone = true;
+
+    public ComposeFragment() {
+        // Required empty public constructor
+    }
+
+    public static ComposeFragment newInstance(Restaurant restaurant) {
+        ComposeFragment fragment = new ComposeFragment();
+        Bundle args = new Bundle();
+        args.putParcelable(ARG_RESTAURANT, Parcels.wrap(restaurant));
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            mRestaurant = (Restaurant) Parcels.unwrap(getArguments().getParcelable(ARG_RESTAURANT));
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // binding.
+        binding = FragmentComposeBinding.inflate(getLayoutInflater(), container, false);
+        View view = binding.getRoot();
+        spinner = binding.simpleSpinner;
+
+        if (mRestaurant != null) {
+            ratingListener(true);
+        } else {
+            ratingListener(false);
+        }
+
+        return view;
+    }
+
+    private void ratingListener(boolean enable) {
+        //if restaurant is selected let user use the rating bar else show message
+        if (enable == true) {
```

Comment:
In this case, `enable` is a `boolean`, so you could just use `if (enable)`.  Something that you could do to make this a bit more clear is to change the param name from `enable` to `isEnabled`, then `if (isEnabled` makes more sense, semantically :-)